server/api/api/endpoints/projects_v2.py

```diff
@@ -137,6 +137,10 @@ def _get_or_create_project_deletion_background_task(
            4. Finish LeaderDeletionJob
         4. Finish MLRunDeletionWrapperTask
     """
+    # The project deletion wrapper should wait for the project deletion to complete. This is a backwards compatibility
+    # feature for when working with iguazio <= 3.5.4 that does not support background tasks and therefore doesn't wait
+    # for the project deletion to complete.
+    wait_for_project_deletion = True
```

Comment:
wait_for_project_deletion should True when igz <= 3.5.4 and request is not from leader.